### PR TITLE
 Accept all typedoc options + simplify usage and the code

### PR DESCRIPTION
**Pull request by mucsi96**
Thanks for creating this very useful loader! I tried to use in, but I faced some issues with the `mode`. I find the `file` mode not good for my [use case](https://nightwatch-api.netlify.com/api/). Mode `modules` works much better as it provides more information about the type definitions. 

With this PR I did:
- Added support for any TypeDoc `mode` 
- Removed the need for `json-loader`
- Improved the `README.md`
- Updated / fixed the dependencies
- Simplified the code by using only one file
- Simplified the test by returning a promise and using the Jest snapshot feature

I hope you like the changes and we can merge this PR. After that I can start using it :) And I hope it can be more useful for others as well.

Let me know if you don't agree with some changes I made.

### CODE REVIEW

**Comment by mucsi96** [comment]
@andrewbranch Can you take a look on my PR?

File: lib/loader.js [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
+  const result = app.converter.convert([this.resourcePath]);
+
+  if (result.errors && result.errors.length) {
+    result.errors.forEach(error => {
+      throw new Error(error.messageText);
+    });
+  }
+
+  const projectObject = app.serializer.projectToObject(result.project);
+  return `export default ${ JSON.stringify(projectObject) }`;
```
Comment (andrewbranch) [inline lib/loader.js]: Let’s be consistent with what json-loader does if we're going to incorporate its functionality:

```suggestion
  return `module.exports = ${JSON.stringify(projectObject)};`;
```

File: README.md [hunk 2]
```diff
@@ -7,12 +7,11 @@ A Webpack loader that converts TypeScript to [TypeDoc](http://typedoc.org) JSON.
...
 module: {
   rules: [{
-    test: /\.md$/,
+    test: /\.ts$/,
     use: {
       loader: 'typedoc-loader',
       options: {
-        // All options are passed to typedoc, except `mode` which is always `file`,
-        // and `json` which is always `true`.
+        // All options are passed to typedoc.
```
Comment (andrewbranch) [inline README.md]: What's the default `mode`?

**Comment by andrewbranch** [comment]
Also—my apologies for not seeing this sooner. Somehow I completely missed the original notification, and only saw it when you just now commented. 😱

**Comment by andrewbranch** [comment]
Closing as housekeeping. Please rebase, reopen, and address feedback if you're still interested. Thanks!

**Pull request closed without merging**